src/react/account/AccountRoleSelectButtonAndModal.tsx

```diff
@@ -18,10 +18,12 @@ function AccountRoleSelectButtonAndModal() {
   const accounts = useAccounts();
   const { path } = useRouteMatch();
   const { account, selectAccountAndRoleRedirectTo } = useCurrentAccount();
-  const accountName = account?.Name;
   const { roleArn } = useDataServiceRole();
   const [assumedRoleArn, setAssumedRoleArn] = useState(roleArn);
-  const [currentAccountName, setCurrentAccountName] = useState(accountName);
+  const [currentAccountName, setCurrentAccountName] = useState(account?.Name);
+  useEffect(() => {
+    if (account?.Name) setCurrentAccountName(account?.Name);
+  }, [JSON.stringify(account)]);
```

Comment:
I'm not sure why an effect is used here ? Can't `currentAccountName` just be simply derived from account ?

```suggestion
  const currentAccountName = account?.Name;
```